### DISCUSSION TITLE
Welcome to Ticker Discussions!

### DISCUSSION THREAD

**Discussion by sstaub** [sstaub/Ticker#30] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by keepinautomacao** [comment]
Hi... During some tests using microseconds precision, I noticed that the output was not respecting the parameterized in the Ticker function.

I made a small change in the library (Ticker.cpp) and it was working as needed.

The change made was as follows:

Ticker::Ticker(fptr callback, uint32_t timer, uint32_t repeat, resolution_t resolution) {
	this->resolution = resolution;
	//if (resolution == MICROS) timer = timer * 1000;
	this->timer = timer;
	this->repeat = repeat;
	this->callback = callback;
	enabled = false;
	lastTime = 0;
	counts = 0;
	}

I only commented one line ( //if (resolution == MICROS) timer = timer * 1000; ) and for me it worked perfectly.

Is this change correct?

**Reply by sstaub** [reply]
can you try
`if (resolution == MICROS) this->timer = timer * 1000;`

**Reply by sstaub** [reply]
Can you give me an example code?
